Fix SD Print Timer

Kaibob2 opened the pull request:
According to 
[Original thread](https://github.com/MarlinFirmware/Marlin/commit/880cdb553b74a0d142236c5c0e03e29cb93ecab3#commitcomment-18731110)

I'm not sure if it is correctly coded, but it works.

Kaibob2 committed:
Fix print duration timer on LCD

Kaibob2 committed:
Fix print duration timer on LCD

I also added the percentage value beside to the left of the SD print timer. Uncomment the three lines below // Percent complete if you don't like it.

thinkyhead commented on Marlin/duration_t.h:
```diff
@@ -146,11 +146,13 @@ struct duration_t {
+        m = this->minute() % 60;
     if (with_days)
-      sprintf_P(buffer, PSTR("%id %02i:%02i"), this->day(), this->hour() % 24, m);
+      sprintf_P(buffer, PSTR("%id %02i:%02i"), d, h, m);
```
Skipping temporary variables usually reduces code size a little bit.

``` cpp
sprintf_P(buffer, PSTR("%id %02i:%02i"), (int)this->day(), (int)(this->hour() % 24), m);
```

thinkyhead commented:
The update of graphical displays is one of the things slowing the potential speed of deltas and other types of machines, and harming the performance of Marlin compared to some other firmwares. This is a fine change in the sense of making the display more informative, but I'm not sure we can afford to keep adding more overhead in this part of the code. Not unless we can find ways to optimize drawing of the text and graphics on these displays generally.

Kaibob2 commented:
I see. So better leave the percentage thing out completely. I didn't take the speed decrease into account. The inormation gain is not that big anyway.

Kaibob2 commented:
Is 
`sprintf_P(buffer, PSTR("%id %02i:%02i"), (int)this->day(), (int)(this->hour() % 24), m);`
the same as what i did?
Wouldn't it have to be
`sprintf_P(buffer, PSTR("%id %02i:%02i"), (int)this->day() % 365, (int)(this->hour() % 24), m);`
then?

Btw.: I haven't verified that days get shown in any case. I never had a printjob exceeding 15 hours. So i have no idea how to test this.

thinkyhead commented:
> the same as what i did?

As I pointed out, using auto variables tends to encourage the compiler to generate a few more bytes of code than it would otherwise. This is not always the case, but if temporaries are not truly needed then they are best avoided.

> Wouldn't it have to be…?

Obviously prints won't go over 365 days, but if they did, omitting the `% 365` will allow the display to show 366 days and higher.

thinkyhead commented:
> i have no idea how to test this.

``` diff
  /**
   * @brief Constructor
   */
  duration_t()
-   : duration_t(0) {};
+   : duration_t(15552000) {};
```

Kaibob2 committed:
Update duration_t.h

Kaibob2 committed:
Update ultralcd_impl_DOGM.h

Kaibob2 commented:
```
  duration_t()
   : duration_t(15552000) {};
```

does nothing.

I changed the code according to your suggestions. It seems to work like this, the days function is still untested. I removed the percent complete too. The position change is done for the timer to line up with the SD progress bar.
I'm sorry, but at this point i'm out. I think i'm not a big help as my coding skills aren't good enough. I think you've got better things to do than correcting every line. Thanks for your great support!
The picture below is with percent and with duration_t(15552000) {}
![20160830_214353_resized](https://cloud.githubusercontent.com/assets/20832276/18104742/d31fdb7e-6efc-11e6-9471-167f9b306c62.jpg)

thinkyhead commented:
> does nothing.

That was the general idea. A quick search of the code shows that actually you can test the days output by changing line 187 of `printcounter.cpp` to something like…

``` cpp
      this->lastDuration = 436500;
```

thinkyhead commented:
Superseded by new PRs - still attributed to KaiBob2.
